ui-v2/app/utils/form/builder.js

```diff
@@ -56,9 +56,11 @@ export default function(changeset = defaultChangeset, getFormNameProperty = pars
         _children[child.getName()] = child;
         return this;
       },
-      handleEvent: function(e) {
+      handleEvent: function(e, targetName) {
         const target = e.target;
-        const parts = getFormNameProperty(target.name);
+        // currently we only use targetName in {{form-component}} for handling deeply
+        // nested forms, once {{form-component}} handles deeply nested forms targetName can go
+        const parts = getFormNameProperty(targetName || target.name);
```

Comment:
I guess I'm a bit confused in isolation why we have to support the additional parameter here – not blocking but feels prone to potential confusion.

Reply:
Yeah your right, ideally it wouldn't be there, but I was trying to think of the least heavy handed approach to fix this, and this was the best one I could think of. There are at least some tests around the thing that broke now, and I will probably add more. So if/when this gets revisited I can make a larger change than potentially touches more things.